### PR TITLE
create and pay, with a few type solving on purchase

### PR DESCRIPTION
**Pull request by DarrylEgbertNangoi-GitHub**
Create:
https://jam.dev/c/92c315ed-c25f-4876-8d4a-440c9a9cc86c

Create and Pay: 
https://jam.dev/c/545352ae-109b-4a1d-849f-7794f1e24822

Update:
https://jam.dev/c/9b74cb1f-6044-49eb-85e3-070a76ed56e5

Update with Pay:
https://jam.dev/c/709b91c6-6114-4393-b5d7-b5e41d4dbdd1

Make a copy:
https://jam.dev/c/ac870c35-f657-45d7-8518-b519f6df7dc7

### CODE REVIEW

File: app/(protected)/expenses/[...expense]/components/ExpenseAddItemsTable.tsx [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+                                  field.onChange(Number(e.target.value));
+                                }}
+                              />
+                              {errors.details?.[idx]?.amount && (
+                                <p className="text-red-500">
+                                  Price must not be empty
+                                </p>
+                              )}
+                            </FormItem>
+                          )}
```
Comment (LeroyMichael) [inline app/(protected)/expenses/[...expense]/components/ExpenseAddItemsTable.tsx:125]: https://github.com/LeroyMichael/PowerUp/blob/b243c3bb26fc7f1f5929a4050f95a1d5f396649e/app/(protected)/transactions/transaction-form/%5B%5B...transaction%5D%5D/page.tsx#L556-L565

perlu tambahin formatting ini dibwh pricenya supaya keliatan cth Rp100.000

**Review by LeroyMichael — CHANGES_REQUESTED** [review]

File: app/(protected)/expenses/[...expense]/components/ExpenseAddItemsTable.tsx (hunk 1 above)
Comment (DarrylEgbertNangoi-GitHub) [reply app/(protected)/expenses/[...expense]/components/ExpenseAddItemsTable.tsx:135]: updated

**Comment by LeroyMichael** [comment]
image
contactnya cba ikutin di page sales deh, jd klo mau update dia yg selected lansgung taro diatas

**Review by LeroyMichael — APPROVED** [review]
need some improvements for contact details
merging the existing code

**Pull request merged**